android/databases/src/main/java/com/coinbase/wallet/libraries/databases/db/Database.kt

```diff
@@ -186,51 +149,48 @@ class Database<T : RoomDatabaseProvider>() {
      * @return A Single wrapping the item returned by the fetch.
      */
     @Suppress("UNCHECKED_CAST")
-    inline fun <reified T : DatabaseModelObject> fetchOne(query: String, vararg args: Any): Single<Optional<T>> {
-        return this.fetch<T>(query, *args).map { it.firstOrNull().toOptional() }
-    }
+    inline fun <reified T : DatabaseModelObject> fetchOne(
+        query: String,
+        vararg args: Any
+    ): Single<Optional<T>> = fetch<T>(query, *args)
+        .map { records ->
+            if (records.count() > 1) {
```

Comment:
Good to have this, but I notice we have places in the code where we use `fetchOne` without supplying `LIMIT 1` to the raw query.
Namely: 
```
AddressDao.getAddress(
        blockchain: Blockchain,
        currencyCode: CurrencyCode,
        network: Network,
        address: String
    ): Single<Optional<Address>> {

WalletDao.getWallet(
        blockchain: Blockchain,
        currencyCode: CurrencyCode
    ): Single<Optional<Wallet>> = database.fetchOne(
        "SELECT * FROM Wallet where blockchain = ? AND currencyCode = ?",
        blockchain.rawValue,
        currencyCode.rawValue
    )

ERC20Dao.getERC20(
        contractAddress: String,
        chainId: Int
    ): Single<Optional<ERC20>> = database.fetchOne(
        "SELECT * FROM ERC20 where contractAddress = ? and chainId = ?",
        contractAddress,
        chainId
    )
```
and
```
DappDao.getDapp(
        url: URL
    ): Single<Optional<Dapp>> 
```

Reply:
`limit 1` is not necessary if the lookup is using a unique index. I added this so we can catch instances where we're returning multiple records